MIGRATING.md

```diff
@@ -2,6 +2,240 @@
  
 This SDK uses semantic versioning and follows Okta's [library version policy](https://developer.okta.com/code/library-versions/). In short, we do not make breaking changes unless the major version changes!
 
+## Migrating from 2.x.x to 3.0.0
+
+Version 3.0.0 of this SDK introduces a number of breaking changes from previous versions. 
+In addition to many new classes/interfaces, some existing classes/interfaces are no longer backward compatible due to method renaming and signature changes.
+
+[Custom HOTP Factor](https://developer.okta.com/docs/reference/api/factors/#enroll-custom-hotp-factor) is supported now.
+Define a separate Policy model for Authorization Servers.
+Define a separate Policy Rule model for Authorization Servers.
+[SAML Multiple ACS URLs](https://developer.okta.com/docs/reference/api/apps/#acs-endpoint-object) is supported now.
+
+### Package `com.okta.sdk.client.Client`
+
+Below method has been added.
+- `void deleteIdentityProviderKey(String keyId)`
+   
+Below methods have undergone a signature change.
+- `listGroups(String q, String filter)` signature changed to `listGroups(String q, String filter, String expand)`
+    - New param `expand` has been added. It's optional and would need to be specified only while using search queries. 
+
+- `listPolicies(String s)` return type changed from `com.okta.sdk.resource.policy.PolicyList` to `com.okta.sdk.resource.authorization.server.AuthorizationServerPolicyList`
```

Comment:
Would be better to give the variable a meaningful name instead of `String s`? replace `s` with a contextually meaningful name.

Reply:
Updated